tests/functional/commands/conftest.py

```diff
@@ -95,19 +99,194 @@ def database():
 
 
 @pytest.fixture()
-def clean_db(database, exp_config):
+def clean_db(database, db_instance):
     """Clean insert example experiment entries to collections."""
     database.experiments.drop()
+    database.lying_trials.drop()
     database.trials.drop()
     database.workers.drop()
     database.resources.drop()
 
 
 @pytest.fixture()
-def only_experiments_db(database, exp_config):
+def db_instance(null_db_instances):
+    """Create and save a singleton database instance."""
+    try:
+        db = Database(of_type='MongoDB', name='orion_test',
+                      username='user', password='pass')
+    except ValueError:
+        db = Database()
+
+    return db
+
+
+@pytest.fixture
+@pytest.mark.usefixtures('clean_db')
+def only_experiments_db(exp_config):
     """Clean the database and insert only experiments."""
-    database.experiments.drop()
     database.experiments.insert_many(exp_config[0])
-    database.trials.drop()
-    database.workers.drop()
-    database.resources.drop()
+
+
+def ensure_deterministic_id(name, db_instance, parent=None, root=None):
+    """Change the id of experiment `name` to `name`."""
+    experiment = db_instance.read('experiments', {'name': name})[0]
+    db_instance.remove('experiments', {'_id': experiment['_id']})
+    experiment['_id'] = name
+    experiment['refers']['root_id'] = name
+
+    if parent is not None and root is None:
```

Comment:
That's a bit difficult to understand. You should document the args in the docstring. Also, should be `elif` I guess.